source/Symbol/SwiftASTContext.cpp

```diff
@@ -99,34 +99,32 @@
 #include "Plugins/SymbolFile/DWARF/DWARFASTParserSwift.h"
 
 #ifdef LLDB_CONFIGURATION_DEBUG
-#define VALID_OR_RETURN(value)                                                 \
+#define _VALID_OR_RETURN(_return)                                              \
   do {                                                                         \
-    lldbassert(!HasFatalErrors());                                             \
     if (HasFatalErrors()) {                                                    \
-      return (value);                                                          \
+      DiagnosticManager diagnostic_manager;                                    \
+      CopyDiagnostics(diagnostic_manager);                                    \
+      llvm::errs() << diagnostic_manager.GetString();                          \
```

Comment:
Ok, so there are two separate issues:

1. Spamming `errs()`: I'm not too sure this is such a big deal as every failed `lldbassert()` already generates lots of output: https://github.com/apple/swift-lldb/blob/765c9aab8edc3b2e30bb0505aee8464c62bbfe30/source/Utility/LLDBAssert.cpp#L26-L31.
2. stderr getting lost: I'm looking at this patch as as attempt to produce more verbose `lldbassert()` for `SwiftASTContext.cpp`, hence logging to the same place where `lldbassert()` logs to. Logging _just_ to a log channel will split some of the information from the rest, and doesn't seem right. Logging to both `errs()` and a log channel is OK, but produces even more spam (and works against 1.)

Ultimately I'm happy with whatever you think is the best here. I just want a way of seeing this information without maintaining a fork. Please let me know which way you want to go, and I'll implement it so.